types/react/test/index.ts

```diff
@@ -248,6 +248,35 @@ const FunctionComponent4: React.FunctionComponent = props => null;
 // leaving here to document limitation and inspect error message
 const FunctionComponent5: React.FunctionComponent = () => false; // $ExpectError
 
+function VoidFunctionComponent(props: SCProps) {
+    return props.foo ? DOM.div(null, props.foo) : null;
+}
+
+// tslint:disable-next-line:no-namespace
+namespace VoidFunctionComponent {
+    export const displayName = "VoidFunctionComponent";
+    export const defaultProps = { foo: 42 };
+}
+
+const VoidFunctionComponent2: React.VoidFunctionComponent<SCProps> =
+    // props is contextually typed
+    props => DOM.div(null, props.foo);
+VoidFunctionComponent2.displayName = "VoidFunctionComponent2";
+VoidFunctionComponent2.defaultProps = {
+    foo: 42
+};
+
+const VoidFunctionComponent3: React.VoidFunctionComponent<SCProps> =
+    // undesired: Rejects implicit usage of props.children
```

Comment:
Since this component isn't supposed to accept children wouldn't the error be desired?

Reply:
Ah yes, sorry about the confusing verbiage here